bazel/p4_library.bzl

```diff
@@ -2,27 +2,73 @@
 
 load("@bazel_tools//tools/cpp:toolchain_utils.bzl", "find_cpp_toolchain")
 
-def _p4_library_impl(ctx):
-    p4c = ctx.executable._p4c
+def _extract_common_p4c_args(ctx):
+    """Extract common arguments for p4c build rules."""
     p4file = ctx.file.src
     p4deps = ctx.files._p4include + ctx.files.deps
-    target = ctx.attr.target
     args = [
         p4file.path,
         "--std",
         ctx.attr.std,
-        "--target",
-        (target if target else "bmv2"),
         "--arch",
         ctx.attr.arch,
     ]
-    if ctx.attr.extra_args:
-        args.append(ctx.attr.extra_args)
 
     include_dirs = {d.dirname: 0 for d in p4deps}  # Use dict to express set.
     include_dirs["."] = 0  # Enable include paths relative to workspace root.
     args += [("-I" + dir) for dir in include_dirs.keys()]
 
+    return args
+
+def _extract_p4c_inputs(ctx):
+    """Extract input p4 files to give to p4c from the build rule context."""
+    p4file = ctx.file.src
+    p4deps = ctx.files._p4include + ctx.files.deps
+    return p4deps + [p4file]
+
+def _run_shell_cmd_with_p4c(ctx, command, **run_shell_kwargs):
+    """Run given sequence of shell commands using `run_shell` action after
```

Comment:
"Runs the given shell command" (since it is only a single string, rather than a list of strings)

Reply:
done